provider/image.go

```diff
@@ -418,11 +418,22 @@ func marshalCachedImages(b resource.PropertyValue) ([]string, error) {
 
 	// if we specify a list of stages, then we only pull those
 	cacheFrom := c.ObjectValue()
+	if _, ok := cacheFrom["images"]; !ok {
+		return cacheImages, fmt.Errorf("cacheFrom requires an `images` field")
+	}
 	if cacheFrom["images"].IsNull() {
-		return cacheImages, fmt.Errorf("if you want to use cacheFrom, you must have `images` set")
```

Comment:
i'm a bit confused about the distinction here between the `cacheFrom["images"]` property being unset as in the prior if-statement check, and here where it could be set but it's a Null type value. would it be an error in both cases? please correct my misunderstanding

Reply:
yeah! so the first check asks "did you set this field? if you haven't, we'll error out and ask you to set it"
the second check says "is the _value_ of `images` null; if so, we need to allow it because it may be an unknown" - this covers the use case where a list may be passed in from another resource output.
Prior to adding this verification to `Check()`, we didn't need to worry about this, because we only called this function from Create(), and Create handles unknowns differently.
I like the early verification, though.